src/impy/models/pythia6.py

```diff
@@ -66,9 +66,9 @@ def children(self):
     @property
     def _charge_init(self):
         if self.charge_vec is None:
-            self.charge_vec = [
+            self.charge_vec = np.asarray([
                 self.lib.pychge(self.lib.pyjets.k[i, 1]) / 3 for i in range(self.npart)
-            ]
+            ])
```

Comment:
Perhaps use np.fromiter to avoid creating a temporary array.

Reply:
OK! Thanks